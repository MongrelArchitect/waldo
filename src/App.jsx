import React, { useState } from 'react';
import Header from './components/Header';
import Game from './components/Game';
import beastie from './images/beastie.jpg';
import creeper from './images/creeper.jpg';
import superStar from './images/super-star.jpg';

export default function App() {
  const [characters, setCharacters] = useState([
    {
      name: 'beastie',
      img: beastie,
      found: false,
    },
    {
      name: 'creeper',
      img: creeper,
      found: false,
    },
    {
      name: 'super star',
      img: superStar,
      found: false,
    },
  ]);

  return (
    <div className="container">
      <Header characters={characters} setCharacters={setCharacters} />
      <Game characters={characters} />
    </div>
  );
}
