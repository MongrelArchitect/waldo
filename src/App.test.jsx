import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
