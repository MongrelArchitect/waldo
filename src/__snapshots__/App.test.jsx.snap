// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <header>
      <nav>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          class=""
          href="/play"
        >
          Play
        </a>
        <a
          class=""
          href="/scores"
        >
          Scoreboard
        </a>
      </nav>
      <span />
      <span
        data-testid="characters"
      >
        <img
          alt="beastie"
          class="not-found"
          src="beastie.jpg"
        />
        <img
          alt="creeper"
          class="not-found"
          src="creeper.jpg"
        />
        <img
          alt="super star"
          class="not-found"
          src="super-star.jpg"
        />
      </span>
    </header>
    <div
      class="game"
      data-testid="game"
    >
      <div
        data-testid="popup"
      >
        <span>
          <img
            alt="beastie"
            src="beastie.jpg"
          />
          beastie
        </span>
        <span>
          <img
            alt="creeper"
            src="creeper.jpg"
          />
          creeper
        </span>
        <span>
          <img
            alt="super star"
            src="super-star.jpg"
          />
          super star
        </span>
      </div>
      <img
        alt="Wimmelbild"
        class="wimmel"
        src="wimmel.jpg"
      />
    </div>
  </div>
</div>
`;
